packages/core/src/tracing/instrumentation/protocols/nativeFetch.js

```diff
@@ -96,15 +95,19 @@ function instrument() {
       originalArgs[i] = arguments[i];
     }
 
-    if (skipTracingResult.skip) {
+    if (
+      skipTracingResult.skip ||
+      (!skipTracingResult.allowRootExitSpan && !parentSpan) ||
+      (!skipTracingResult.allowRootExitSpan && skipTracingResult.isExitSpan)
+    ) {
       if (skipTracingResult.suppressed) {
         injectSuppressionHeader(originalArgs, w3cTraceContext);
       }
       return originalFetch.apply(originalThis, originalArgs);
     }
 
     return cls.ns.runAndReturn(() => {
-      const span = cls.startSpan('node.http.client', constants.EXIT, parentSpan.t, parentSpan.s);
+      const span = cls.startSpan('node.http.client', constants.EXIT, parentSpan?.t, parentSpan?.s);
```

Comment:
LGTM 👍 
Could you pls add a comment:

// NOTE: We need to check for parentSpan existence, because of `allowExitRootSpan` being true

Reply:
@abhilash-sivan

Reply:
done